ee/clickhouse/queries/event_query.py

```diff
@@ -95,7 +100,7 @@ def _determine_should_join_persons(self) -> None:
 
         # :KLUDGE: The following is mostly making sure if cohorts are included as well.
         #   Can be simplified significantly after https://github.com/PostHog/posthog/issues/5854
-        if any(self._should_property_join_persons(prop) for prop in self._filter.properties):
+        if any(self._should_property_join_persons(prop) for prop in self._filter.property_groups_flat):
```

Comment:
Is property_groups_flat a purely transitional helper or something we'll keep around?

Reply:
It should stick around because it's necessary for digesting what properties are being used so that any functions that are checking on properties being filtered on don't need to keep redoing a tree traversal